sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/implementation/SyncTokenPolicyTest.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.test.TestBase;
+import com.azure.core.test.http.MockHttpResponse;
+import com.azure.core.test.http.NoOpHttpClient;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SyncTokenPolicyTest extends TestBase {
+
+    private static final String SYNC_TOKEN = "Sync-Token";
+    private static final String ID = "jtqGc1I4";
+    private static final String NEW_ID = "newID";
+    private static final String VALUE = "MDoyOA==";
+    private static final String UPDATED_VALUE = "UpdatedValue";
+    private static final long SEQUENCE_NUMBER = 28;
+    private static final long UPDATED_SEQUENCE_NUMBER = 30;
+
+    @Test
+    public void parseSyncToken() {
+        String syncTokenString = constructSyncTokenString(ID, VALUE, SEQUENCE_NUMBER);
+        final SyncToken syncToken = SyncToken.fromSyncTokenString(syncTokenString);
+        syncTokenEquals(syncToken, ID, VALUE, SEQUENCE_NUMBER);
+    }
+
+    @Test
+    public void singleSyncTokenTest() throws Exception {
+        final SyncTokenPolicy syncTokenPolicy = new SyncTokenPolicy();
+
+        final String firstSyncToken = constructSyncTokenString(ID, VALUE, SEQUENCE_NUMBER);
+        final HttpPipeline pipeline = customizedPipeline(firstSyncToken, syncTokenPolicy);
+        final HttpResponse response = pipeline.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
```

Comment:
Step Verifier should be used... I'm not sure why you don't use Mockito to mock an HttpResponse and instead.